src/rtt.c

```diff
@@ -73,45 +81,38 @@ static char xmit_buf[RTT_UP_BUF_SIZE];
 **********************************************************************
 */
 
-uint32_t fastsrch(target *cur_target)
+static uint32_t fast_search(target *cur_target, uint32_t ram_start, uint32_t ram_end)
 {
-	const uint32_t m = 16;
-	const uint64_t q = 0x797a9691; /* prime */
-	const uint64_t rm = 0x73b07d01;
-	const uint64_t p = 0x444110cd;
-	const uint32_t stride = 128;
+	static const uint32_t m = 16;
+	static const uint64_t q = 0x797a9691; /* prime */
+	static const uint64_t r = 0x73b07d01;
+	static const uint64_t p = 0x444110cd;
+	static const uint32_t stride = 128;
 	uint64_t t = 0;
-	uint8_t srch_buf[m+stride];
+	uint8_t srch_buf[m + stride];
 
-	for (struct target_ram *r = cur_target->ram; r; r = r->next) {
-		const uint32_t ram_start = r->start;
-		const uint32_t ram_end = r->start + r->length;
+	t = 0;
```

Comment:
This looks like a redundant assignment as it's already taken care of by line 91?